crates/transaction-emitter-lib/src/transaction_generator/publishing/module_simple.rs

```diff
@@ -0,0 +1,336 @@
+        EntryPoints::Half => get_payload_void(module_id, ident_str!("half").to_owned()),
+        _ => panic!("bogus zero_args_function id generation, check beginning of function"),
+    }
+}
+
+pub fn rand_gen_function(rng: &mut StdRng, module_id: ModuleId) -> TransactionPayload {
+    let fun_idx = rng.gen_range(ZERO_ARG_ENTRY_POINTS_START, ONE_ARG_ENTRY_POINTS_END + 1);
+    match EntryPoints::try_from(fun_idx).expect("Must pick a function in range, bogus id generated")
+    {
+        // 0 args
+        EntryPoints::Nop => get_payload_void(module_id, ident_str!("nop").to_owned()),
+        EntryPoints::Step => get_payload_void(module_id, ident_str!("step").to_owned()),
+        EntryPoints::GetCounter => {
+            get_payload_void(module_id, ident_str!("get_counter").to_owned())
+        }
+        EntryPoints::ResetData => get_payload_void(module_id, ident_str!("reset_data").to_owned()),
+        EntryPoints::Double => get_payload_void(module_id, ident_str!("double").to_owned()),
+        EntryPoints::Half => get_payload_void(module_id, ident_str!("half").to_owned()),
+        // 1 arg
+        EntryPoints::Loopy => loopy(rng, module_id),
+        EntryPoints::GetFromRandomConst => get_from_random_const(rng, module_id),
+        EntryPoints::SetId => set_id(rng, module_id),
+        EntryPoints::SetName => set_name(rng, module_id),
+        _ => panic!("bogus rand_gen_function id generation, check beginning of function"),
+    }
+}
+
+pub fn any_function(
+    rng: &mut StdRng,
+    module_id: ModuleId,
+    other: AccountAddress,
+) -> TransactionPayload {
+    let fun_idx = rng.gen_range(ENTRY_POINTS_START, ENTRY_POINTS_END + 1);
+    match EntryPoints::try_from(fun_idx).expect("Must pick a function in range, bogus id generated")
+    {
+        // 0 args
+        EntryPoints::Nop => get_payload_void(module_id, ident_str!("nop").to_owned()),
+        EntryPoints::Step => get_payload_void(module_id, ident_str!("step").to_owned()),
+        EntryPoints::GetCounter => {
+            get_payload_void(module_id, ident_str!("get_counter").to_owned())
+        }
+        EntryPoints::ResetData => get_payload_void(module_id, ident_str!("reset_data").to_owned()),
+        EntryPoints::Double => get_payload_void(module_id, ident_str!("double").to_owned()),
+        EntryPoints::Half => get_payload_void(module_id, ident_str!("half").to_owned()),
+        // 1 arg
+        EntryPoints::Loopy => loopy(rng, module_id),
+        EntryPoints::GetFromRandomConst => get_from_random_const(rng, module_id),
+        EntryPoints::SetId => set_id(rng, module_id),
+        EntryPoints::SetName => set_name(rng, module_id),
+        // 2 args, second arg existing account address with data
+        EntryPoints::Maximize => maximize(module_id, other),
+        EntryPoints::Minimize => minimize(module_id, other),
+        // 3 args
+        EntryPoints::MakeOrChange => make_or_change(rng, module_id),
+    }
```

Comment:
can you have this indirection in a common function, and then have all these function call it (even if their fun_idx is in the range that is only subset of the functions?)

basically having a single match, instead of maving match duplicated in the code?

Reply:
done, great idea!